framework/src/base/FEProblemBase.C

```diff
@@ -129,6 +129,9 @@ validParams<FEProblemBase>()
                         false,
                         "This causes PETSc to error if it had to reallocate memory in the Jacobian "
                         "matrix due to not having enough nonzeros");
+  params.addParam<bool>("ignore_zeros_in_jacobian",
+                        false,
```

Comment:
PETSc does not ignore zeros by default. If we change this default setting, there are some test fails. BTW, I personally do not like change the default setting in PETSc, since it may bring some unexpected results.